challenge/files/11_maps.toml

```diff
@@ -0,0 +1,141 @@
+]
+
+[[questions]]
+text = """What happens when the number of entries in a Go map exceeds a certain threshold?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "The map allocates a new array with double the capacity and rehashes all keys" },
+    { text = "The map remains the same size but uses a more sophisticated hash function" },
+    { text = "The map splits existing buckets and rehashes only a portion of the keys", code_line_ranges = [] },
+    { text = "The map creates a second, independent map and balances the keys between the two maps" },
+]
+
+[[questions]]
+text = """The load factor of a map is a critical concept in understanding its efficiency. 
+Which of the following best describes the load factor in the context of Go maps?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "The ratio of the number of entries to the total capacity of the map" },
+    { text = "The ratio of the number of used buckets to the total number of buckets" },
+    { text = "The average number of key-value pairs stored in each bucket", code_line_ranges = [] },
+    { text = "The ratio of the number of overflow buckets to the total number of buckets" },
+]
+
+
+[[questions]]
+text = """How does the Go runtime handle memory management and garbage collection for maps, particularly regarding unused keys?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "The Go runtime immediately reclaims memory for a map entry when a key is deleted" },
+    { text = "The Go runtime marks deleted keys as unused, and the memory is reclaimed during the next garbage collection cycle", code_line_ranges = [] },
+    { text = "The Go runtime never reclaims memory for deleted keys; the memory remains allocated until the entire map is garbage collected" },
+    { text = "The Go runtime reclaims memory for deleted keys only when the map is resized" },
+]
+
+
+[[questions]]
+text = """Why is the iteration order of keys in a Go map randomized?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "To improve the performance of map operations by avoiding cache misses" },
+    { text = "To ensure fairness in concurrent map access in multi-threaded programs" },
+    { text = "To prevent programs from relying on a specific iteration order, which could lead to subtle bugs", code_line_ranges = [] },
+    { text = "To enhance security by making it harder to predict the order of key-value pairs" },
+]
+
+[[questions]]
+text = """Which of the following statements about the hash function used in Go maps is true?
+
+"""
+type = "select_answers"
+answers = [
+    { text = "Go uses a simple modulo operation to compute hash values for keys" },
+    { text = "The hash function is deterministic and always produces the same hash for the same input across different runs" },
+    { text = "The hash function includes a per-map random seed to protect against certain types of attacks", code_line_ranges = [] },
```

Comment:
Let's specify the attack name and add proof in the learning links. I don't know about it and want to read)

Reply:
Let's specify "attack when malicious actor will flood hashmap with collisions"

And let's add link in useful links

Once, I was asked about bucket overflow (or something like that) when the attacker tries to find keys to overflow only a single bucket it makes service significantly slower.